### ISSUE TITLE
Plugin Check incorrectly reports %i placeholder as unsupported in WordPress 6.9

### ISSUE THREAD

**Issue by LeninZapata** [10up/plugin-check#1162]
## Description
Plugin Check is reporting an error stating that the `%i` placeholder is "only supported in WP 6.2 or higher" even though I'm running WordPress 6.9.

## Environment
- **WordPress Version:** 6.9
- **Plugin Check Version:** 1.8.0
- **PHP Version:** 8.1

## Steps to Reproduce
1. Create a custom database handler class using `$wpdb->prepare()` with `%i` placeholder
2. Run Plugin Check on the code
3. Plugin Check reports: `WordPress.DB.PreparedSQLPlaceholders.UnsupportedIdentifierPlaceholder - The %i modifier is only supported in WP 6.2 or higher. Found: "%i".`

## Sample Code
```php
global $wpdb;
$table_name = $wpdb->prefix . 'my_custom_table';

// This triggers the error even on WP 6.9
$results = $wpdb->get_results($wpdb->prepare(
    "SELECT * FROM %i WHERE id = %d",
    $table_name,
    123
));
```

## Expected Behavior
Plugin Check should recognize that `%i` is supported since WordPress version is 6.9 (well above the 6.2 requirement).

## Actual Behavior
Plugin Check throws an ERROR stating the placeholder is unsupported, even though:
- WordPress 6.9 is installed
- The `%i` placeholder has been available since WordPress 6.2
- `$wpdb->has_cap('identifier_placeholders')` returns `true`

## Additional Context
The `%i` placeholder was introduced in WordPress 6.2 ([official announcement](https://make.wordpress.org/core/2022/10/08/escaping-table-and-field-names-with-wpdbprepare-in-wordpress-6-1/)) specifically to address the `PreparedSQL.InterpolatedNotPrepared` warnings when working with custom database tables.

It appears Plugin Check may be:
1. Using an incorrect version detection mechanism
2. Testing against a minimum WordPress version declared in plugin headers rather than the actual installed version
3. Not properly checking for `identifier_placeholders` capability

## Workaround
Currently reverting to using interpolated table names with `phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared` comments, which generates warnings but is functionally correct.

## Request
Could you please investigate why Plugin Check is not correctly detecting WordPress version or the availability of the `%i` placeholder? This is causing false positives that prevent developers from using the recommended modern approach for handling table names in prepared statements.

**Comment by ernilambar** [comment]
What is the value of "Requires at least" header in main plugin header which you are testing?

**Comment by LeninZapata** [comment]
> What is the value of "Requires at least" header in main plugin header which you are testing?

Requires at least: 5.8

Note: It's a new plugin, so WordPress recommended I run a test with Plugins Check, but that plugin seems to be giving false positives or something's wrong. I did several SQL handling tests, but nothing was enough to satisfy it. When the plugin is approved, I'll post here so you can review it.

**Comment by dd32** [comment]
> Requires at least: 5.8

@LeninZapata  The `%i` was introduced in WP 6.2, so plugin check is correctly alerting that you're using a placeholder that is not supported in the minimum WordPress version you say the plugin works with. If you increase the minimum WP version to 6.2, this alert should go away.

This would be similar to saying the plugin works with PHP 5.2+ and then using PHP 8 syntax throughout the plugin.

**Comment by LeninZapata** [comment]
Finally my plugin was accepted, however I don't know why pluginCheck identifies that "warning" about the query issue. You can download the plugin at this link https://transfer.it/t/NIUf6ds31JpK, but you should check the file is "class-db-handler.php" (https://pastelink.net/iwckr786)

**Comment by dd32** [comment]
@LeninZapata That plugin version included there doesn't reference `%i`. It complains that you're not using %i and instead using `$table_name` interpolation. 

Adjusting it to use `%i` causes it to flag no issues.
